lexos/processors/analyze/content_analysis_model.py

```diff
@@ -0,0 +1,461 @@
+        """
+
+        :return: a list containing all active dictionaries
+        """
+        return [dictionary for dictionary in self.dictionaries
+                if dictionary.active]
+
+    def count(self):
+        """counts all dictionaries for all active files in the corpus"""
+        self._counters = []
+        dictionaries = self.join_active_dicts()
+        for file in deepcopy(self._corpus):
+            dictionaries = count_phrases(dictionaries, file)
+            self.get_dictionary_counts(dictionaries)
+
+    def get_dictionary_counts(self, dictionaries):
+        """gets the counts for each dictionary
+
+        :param dictionaries: list of Phrase object
+        """
+        counter = []
+        active_dicts = self.get_active_dicts()
+        for dictionary in active_dicts:
+            count = 0
+            for phrase in dictionaries:
+                if phrase.dict_label == dictionary.label:
+                    count += phrase.count
+            counter.append(count)
+            if len(counter) == len(active_dicts):
+                self._counters.append(counter)
+
+    def generate_scores(self):
+        """calculate the formula and scores=formula/total_word_count for each
+        file in the corpus
+
+        """
+        self._scores = []
+        self._formulas = []
+        active_dicts = self.get_active_dicts()
+        result = 0
+        for i in range(len(self._corpus)):
+            new_formula = self._formula
+            for j in range(len(active_dicts)):
+                new_formula = new_formula.replace(
+                    "[" + active_dicts[j].label + "]",
+                    str(self._counters[i][j]))
+            new_formula = new_formula.replace("()", "")
+            try:
+                result = eval(new_formula)
+            except (ValueError, SyntaxError):
+                pass
+            self._scores.append(round(
+                float(result) / self._corpus[i].total_word_counts, 3))
+            self._formulas.append(result)
+
+    def generate_averages(self):
+        """Calculates the averages of eachm dictionary count, formula,
+        total_word_count, and score
+
+        """
+        self._averages = []
+        scores_sum = 0
+        total_word_counts_sum = 0
+        formulas_sum = 0
+        active_dicts = self.get_active_dicts()
+        for index, (score, formula, file) in enumerate(zip(self.scores,
+                                                           self._formulas,
+                                                           self._corpus)):
+            scores_sum += score
+            total_word_counts_sum += \
+                file.total_word_counts
+            formulas_sum += formula
+        if len(self.scores) != 0:
+            scores_avg = round(
+                (float(scores_sum) / len(self.scores)), 3)
+        else:
+            scores_avg = 0
+        if len(self._corpus) != 0:
+            total_word_counts_avg = round((float(total_word_counts_sum) /
```

Comment:
Why 1?

Reply:
After seeing this multiple times, it occurs to me that you are rounding to one decimal place. This is why you need to use keyword parameters, especially with messy function calls like this

Reply:
to round to the first decimal

Reply:
what's messy?

Reply:
In this line you invoke 3 function that is nested in one another, and there is 5 nested brackets. This is line is messy, you may want to split it into 2 lines or add more comments